providers/oidc.go

```diff
@@ -279,27 +287,40 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 	return claims, nil
 }
 
-func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) []string {
+func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) ([]string, error) {
 	groups := []string{}
 
 	rawGroups, ok := rawClaims[p.GroupsClaim].([]interface{})
 	if rawGroups != nil && ok {
 		for _, rawGroup := range rawGroups {
-			group, ok := rawGroup.(string)
-			if ok {
-				groups = append(groups, group)
+			err := appendGroup(&groups, rawGroup)
+			if err != nil {
+				logger.Errorf("unable to append group of type %s with error %s", reflect.TypeOf(rawGroup), err)
 			}
 		}
 	}
 
-	return groups
+	return groups, nil
+}
+
+func appendGroup(groups *[]string, rawGroup interface{}) error {
+	group, ok := rawGroup.(string)
+	if !ok {
+		jsonGroup, err := json.Marshal(rawGroup)
+		if err != nil {
+			return err
+		}
+		group = string(jsonGroup)
+	}
+	*groups = append(*groups, group)
```

Comment:
Lets do the slice appending back up in the caller of this -- and just make this a `formatGroup` helper or something that returns the formatted group `string`